src/Microsoft.AspNetCore.Razor.Evolution/Legacy/CSharpCodeParser.cs

```diff
@@ -295,8 +305,17 @@ private void AfterTransition()
                         }
                         else if (CurrentSymbol.Type == CSharpSymbolType.Keyword)
                         {
-                            KeywordBlock(topLevel: true);
-                            return;
+                            if (TryGetDirectiveHandler(CurrentSymbol.Content, out var handler))
```

Comment:
These directives will be tokenized as keywords, so you need to handle that case.